scripts/py_matter_yamltests/matter_yamltests/errors.py

```diff
@@ -0,0 +1,163 @@
+                message = f'Unknown key: "{key}". Did you mean "{candidate_key}" ?'
+                break
+        super().__init__(message)
+
+        self.tag_key_with_error(content, 'name')
+
+
+class TestStepInvalidTypeError(TestStepError):
+    """Raise when the value for a given key is not of the expected type."""
+
+    def __init__(self, content, key, expected_type):
+        if isinstance(expected_type, tuple):
+            expected_name = ''
+            for _type in expected_type:
+                expected_name += _type.__name__ + ','
+            expected_name = expected_name[:-1]
+        else:
+            expected_name = expected_type.__name__
+        received_name = type(content[key]).__name__
+        message = f'Unexpected type. Expecting "{expected_name}", got "{received_name}"'
+        super().__init__(message)
+
+        self.tag_key_with_error(content, key)
+
+
+class TestStepGroupResponseError(TestStepError):
+    """Raise when a test step targetting a group of nodes expects a response."""
+
+    def __init__(self, content):
+        message = 'Group command should not expects a response'
+        super().__init__(message)
+
+        self.tag_key_with_error(content, 'groupId')
+        self.tag_key_with_error(content, 'response')
+
+
+class TestStepVerificationStandaloneError(TestStepError):
+    """Raise when a test step with a verification key is enabled and not interactive."""
+
+    def __init__(self, content):
+        message = 'Step using "verification" key should either set "disabled: true" or "PICS: PICS_USER_PROMPT"'
+        super().__init__(message)
+
+        self.tag_key_with_error(content, 'verification')
+
+
+class TestStepNodeIdAndGroupIdError(TestStepError):
+    """Raise when a test step contains both "nodeId" and "groupId" keys."""
+
+    def __init__(self, content):
+        message = '"nodeId" and "groupId" are mutually exclusive'
+        super().__init__(message)
+
+        self.tag_key_with_error(content, 'nodeId')
+        self.tag_key_with_error(content, 'groupId')
+
+
+class TestStepValueAndValuesError(TestStepError):
+    """Raise when a test step response contains both "value" and "groupId" keys."""
+
+    def __init__(self, content):
+        message = '"value" and "values" are mutually exclusive'
+        super().__init__(message)
+
+        self.tag_key_with_error(content, 'value')
+        self.tag_key_with_error(content, 'values')
+
+
+class TestStepOptionalMultipleResponsesNotSupported(TestStepError):
+    """Raise when a test step is optional but expects multiple responses."""
+
+    def __init__(self, content):
+        message = 'The "optional" keyword can not be used with multiple expected responses'
+        super().__init__(message)
+
+        self.tag_key_with_error(content, 'optional')
+
+
+class TestStepWaitForMultipleResponsesNotSupported(TestStepError):
+    """Raise when a test step is optional but expects multiple responses."""
```

Comment:
This docstring is wrong, but I am not sure what the right one is.

Reply:
Fixed.